strictdoc/server/routers/main_router.py

```diff
@@ -2696,6 +2698,49 @@ def get_search(q: Optional[str] = None):
             status_code=200,
         )
 
+    # @relation(SDOC-SRS-120, scope=function)
+    @router.get("/autocomplete/uid", response_class=Response)
+    def get_autocomplete_results(q: Optional[str] = None):
```

Comment:
This query should have the second parameter which is the optional MID of the node if this HTTP request is working for an existing node, not a new node.

This parameter shall than be used to ensure that the node's own UID is not highlighted and cannot be inserted.

Without this, I am able to provide a UID of the edited note and an assert is hit when I try to save the node.